src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -112,15 +112,19 @@ public function __construct($name, $value = null, $expire = 0, $path = '/', $dom
         } elseif (!is_numeric($expire)) {
             $expire = strtotime($expire);
 
-            if (false === $expire || -1 === $expire) {
+            if (false === $expire) {
                 throw new \InvalidArgumentException('The cookie expiration time is not valid.');
             }
         }
 
+        if (0 > $expire) {
+            throw new \InvalidArgumentException('The cookie expiration time must be a positive number value.');
```

Comment:
This is a bc break if anyone is doing this for some reason.
We could deprecate, but the behavior is just fine when negative numbers are used here.
Instead, I'd rather fix getMaxAge and turn `0 !== $this->expire` into `0 >= $this->expire`.
and merge this in a lower branch